Promact.Oauth.Server/src/Promact.Oauth.Server/Repository/UserRepository/UserRepository.cs

```diff
@@ -271,558 +145,490 @@ public IEnumerable<UserAc> GetAllUsers()
+        /// This method is used to check email is already exists in database.
         /// </summary>
-        /// <param name="email"></param>
+        /// <param name="email">Passed user email address</param>
         /// <returns> boolean: true if the email exists, false if does not exist</returns>
-        public async Task<bool> CheckEmailIsExists(string email)
+        public async Task<bool> CheckEmailIsExistsAsync(string email)
         {
             var user = await _userManager.FindByEmailAsync(email);
-            return user == null ? false : true;
+            return user != null;
         }
 
 
         /// <summary>
-        /// Fetches user with the given Slack User Id
+        /// Fetch user with given slack user name
         /// </summary>
-        /// <param name="slackUserName"></param>
-        /// <returns></returns>
-        public ApplicationUser FindUserBySlackUserName(string slackUserName)
+        /// <param name="slackUserName">Passed slack user name</param>
+        /// <returns>If user is exists return user otherwise throw SlackUserNotFound exception.</returns>
+        public async Task<ApplicationUser> FindUserBySlackUserNameAsync(string slackUserName)
         {
-            var user = _applicationUserDataRepository.FirstOrDefault(x => x.SlackUserName == slackUserName);
+            var user = await _applicationUserDataRepository.FirstOrDefaultAsync(x => x.SlackUserName == slackUserName);
             if (user == null)
                 throw new SlackUserNotFound();
-            else
-                return user;
+            return user;
         }
 
+
         /// <summary>
-        /// This method used for re -send mail for user credentails
+        /// This method used for re-send mail for user credentials. -An
         /// </summary>
-        /// <param name="id"></param>
+        /// <param name="id">passed user id</param>
         /// <returns></returns>
-        public async Task<bool> ReSendMail(string id)
+        public async Task ReSendMailAsync(string id)
         {
-            _logger.LogInformation("start Resend Mail Method in User Repository");
             var user = await _userManager.FindByIdAsync(id);
             string newPassword = GetRandomString();
-            var code = await _userManager.GeneratePasswordResetTokenAsync(user);
-            IdentityResult result = await _userManager.ResetPasswordAsync(user, code, newPassword);
-            if (result.Succeeded)
-            {
-                _logger.LogInformation("Successfully Reset Password");
-                if (SendEmail(user, newPassword))
-                    return true;
-            }
-            return false;
+            var token = await _userManager.GeneratePasswordResetTokenAsync(user); //genrate passsword reset token
+            IdentityResult result = await _userManager.ResetPasswordAsync(user, token, newPassword);
```

Comment:
This method belong to someone else but still variable was not in use. I have removed it